pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -115,13 +116,28 @@ func (t ResolvedPipelineRunTask) isFailure() bool {
 	}
 	var c *apis.Condition
 	var isDone bool
-	if t.IsCustomTask() {
+
+	switch {
+	case t.IsCustomTask():
 		if t.Run == nil {
 			return false
 		}
 		c = t.Run.Status.GetCondition(apis.ConditionSucceeded)
 		isDone = t.Run.IsDone()
-	} else {
+	case t.IsMatrixed():
```

Comment:
how does this work if a pipeline task is both a custom task and matrixed? Is that out of scope for this PR?

Reply:
yes, that is out of scope here in this PR because it'd be huge otherwise - will add that case when I implement support for matrixed Runs